packages/EasyDecision/src/Decisions/DecisionFactory.php

```diff
@@ -155,6 +170,8 @@ static function (DecisionConfiguratorInterface $first, DecisionConfiguratorInter
             $configurator->configure($decision);
         }
 
+        $this->decisionAggregator->addDecision($decision, $configurators);
```

Comment:
Even after removing the aggregator make sure that you collect the configurators for a decision within the foreach above because as you can see configurator can implement the `RestrictedDecisionConfiguratorInterface` and they might not support the given decision so they shouldn't be in the array for that decision.

Reply:
haha this was a mistake, Added tests to make sure it's working as expected